endpoints/github.ts

```diff
@@ -124,9 +120,15 @@ async function status ({ owner, repo, ref = 'master', context }: Args) {
     : resp!.state
 
   if (Array.isArray(state)) {
+    const error = state.find(x => x.state === 'error')
     const failure = state.find(x => x.state === 'failure')
     const pending = state.find(x => x.state === 'pending')
-    state = failure ? 'failure' : (pending ? 'pending' : 'success')
+
+    if (error) {
```

Comment:
pretty ugly one, but... meeh.

Reply:
Maybe create a function for combining states:
```
function combineState (states) {
	if (states.find(x => x.state === 'error')) return 'error'
	if (states.find(x => x.state === 'failure')) return 'failure'
	if (states.find(x => x.state === 'pending')) return 'pending'
	if (states.every(x => x.state === 'success')) return 'success'

	// this shouldn't happen, but in case it happens
	throw new Error(`Unknown states: ${states.map(x => x.state).join()}`)
}
```

Reply:
Great! Good point :)